java-frontend/src/main/java/org/sonar/java/resolve/JavaSymbol.java

```diff
@@ -522,6 +523,61 @@ public MethodJavaSymbol(int flags, String name, JavaSymbol owner) {
       this.typeVariableTypes = Lists.newArrayList();
     }
 
+    public String completeSignature(){
+      String sign = "";
+      if(owner != null) {
+        sign += owner.getType().fullyQualifiedName();
+      }
+      sign += "#" + name + desc();
+      return sign;
+    }
+
+    private String desc() {
+      if(desc == null) {
+        org.objectweb.asm.Type ret = returnType == null ? org.objectweb.asm.Type.VOID_TYPE : toAsmType(returnType.type);
+        org.objectweb.asm.Type[] argTypes = new org.objectweb.asm.Type[0];
+        if(super.type != null) {
+          argTypes = getParametersTypes().stream().map(MethodJavaSymbol::toAsmType).toArray(org.objectweb.asm.Type[]::new);
+        }
+        desc = org.objectweb.asm.Type.getMethodDescriptor(ret, argTypes);
+      }
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS106)